go-controller/pkg/clustermanager/endpointslice_mirror_controller_test.go

```diff
@@ -0,0 +1,445 @@
+
+				_, err := fakeCM.fakeClient.NetworkAttchDefClient.K8sCniCncfIoV1().NetworkAttachmentDefinitions(namespaceT.Name).Create(
+					context.TODO(),
+					testing.GenerateNAD("l3-network", "l3-network", namespaceT.Name, types.Layer3Topology, "10.128.2.0/16/24", types.NetworkRoleSecondary),
+					metav1.CreateOptions{})
+				gomega.Expect(err).ToNot(gomega.HaveOccurred())
+
+				gomega.Eventually(func() error {
+					// defaultEndpointSlice should exist
+					_, err := fakeCM.fakeClient.KubeClient.DiscoveryV1().EndpointSlices(namespaceT.Name).Get(context.TODO(), defaultEndpointSlice.Name, metav1.GetOptions{})
+					if err != nil {
+						return err
+					}
+					return nil
+				}).ShouldNot(gomega.HaveOccurred())
+
+				gomega.Consistently(func() error {
+					// no mirrored EndpointSlices should exist
+					mirrorEndpointSliceSelector := labels.Set(map[string]string{
+						discovery.LabelManagedBy: types.EndpointSliceMirrorControllerName,
+					}).AsSelectorPreValidated()
+
+					mirroredEndpointSlices, err := fakeCM.fakeClient.KubeClient.DiscoveryV1().EndpointSlices(namespaceT.Name).List(context.TODO(), metav1.ListOptions{LabelSelector: mirrorEndpointSliceSelector.String()})
+					if err != nil {
+						return err
+					}
+					if len(mirroredEndpointSlices.Items) != 0 {
+						return fmt.Errorf("expected no mirrored EndpointSlices")
+					}
+					return nil
+				}).WithTimeout(5 * time.Second).ShouldNot(gomega.HaveOccurred())
+
+				return nil
+			}
+
+			err := app.Run([]string{app.Name})
+			gomega.Expect(err).ToNot(gomega.HaveOccurred())
+		})
+
+		ginkgo.It("should update/delete mirrored EndpointSlices in namespaces that use user defined networks as primary ", func() {
+			app.Action = func(ctx *cli.Context) error {
+				namespaceT := *newNamespace("testns")
+
+				pod := v1.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:        "test-pod",
+						Namespace:   namespaceT.Name,
+						Annotations: map[string]string{util.OvnPodAnnotationName: `{"default":{"mac_address":"0a:58:0a:f4:02:03","ip_address":"10.244.2.3/24","role":"infrastructure-locked"},"testns/l3-network":{"mac_address":"0a:58:0a:80:02:04","ip_address":"10.128.2.4/24","role":"primary"}}`},
+					},
+					Status: v1.PodStatus{Phase: v1.PodRunning},
+				}
+
+				defaultEndpointSlice := discovery.EndpointSlice{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "default-endpointslice",
+						Namespace: namespaceT.Name,
+						Labels: map[string]string{
+							discovery.LabelServiceName: "svc2",
+							discovery.LabelManagedBy:   types.EndpointSliceDefaultControllerName,
+						},
+						ResourceVersion: "1",
+					},
+					Endpoints: []discovery.Endpoint{
+						{
+							Addresses: []string{"10.244.2.3"},
+							TargetRef: &v1.ObjectReference{
+								Kind:      "Pod",
+								Namespace: namespaceT.Name,
+								Name:      pod.Name,
+							},
+						},
+					},
+				}
+				mirroredEndpointSlice := mirroredEndpointSlice(&defaultEndpointSlice, "l3-network")
```

Comment:
nit: the variable name and the method name are the same. It confused me when reviewing. 

Maybe consider re-naming one of them - possible naming the method `mirrorEndpointSlice` is more accurate (since it is a verb).